install/installer/pkg/components/ws-manager-mk2/deployment.go

```diff
@@ -176,7 +176,7 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 			},
 			Spec: appsv1.DeploymentSpec{
 				Selector: &metav1.LabelSelector{MatchLabels: labels},
-				Replicas: common.Replicas(ctx, Component),
+				Replicas: pointer.Int32(2),
```

Comment:
This could cause issues with activity timeouts. The load balancer will not know which instance is the leader so workspace activity could be reported to the standby and workspaces will time out. I think in practice it will not be a problem because we send heartbeats often enough that the leader will still get notified before that happens. Something to be aware of though.

Reply:
That's a good point. I think we should find a solution for that before going with 2 replicas, otherwise it will become hard to investigate any timeout issues

Reply:
> That's a good point. I think we should find a solution for that before going with 2 replicas, otherwise it will become hard to investigate any timeout issues

Can you expand what do you mean by timeout issues?

Reply:
> The load balancer only sees one instance, not two

Why does it see only one? If we have two replicas would it not send requests to all two? Does the standby not become ready?

> Can you expand what do you mean by timeout issues?

The workspaces send activity requests to ws-manager in order to mark the workspace as active. We store this information in memory in ws-manager. If the requests would go to the standby the active manager (which checks the timeouts) would not have the information that the workspace is active because that information is stored by the replica.

Reply:
Correct, the standby is not ready to serve, but passing the ready health checks.

You can see this in the preview killing the current leader while tailing the other pod logs

Reply:
> We store this information in memory in ws-manager

This is wrong if when we switch the leader the other pod starts killing workspaces without waiting for status reports from workspaces

Reply:
There's a callback we can implement when a pod is elected the new leader and wait for workspace reports before change anything